docs/API.md

```diff
@@ -1,12 +1,237 @@
+<dd><p>Turns URL parameters into key-value pairs. Assigns type
+to boolean and number values.</p>
+</dd>
+<dt><a href="#ispair">ispair(e)</a> ⇒ <code>boolean</code></dt>
+<dd><p>Checks if the argument is a pair (array with two elements, the first
+must be a string)</p>
+</dd>
+<dt><a href="#flat">flat(arr)</a> ⇒ <code>array</code></dt>
+<dd><p>Flattens arrays of arrays (only one level) so that
+an array af pairs remains</p>
+</dd>
+<dt><a href="#unnest">unnest(obj, prefix)</a> ⇒ <code>Array</code></dt>
+<dd><p>Turns a nested QBL in JSON notation into a list of
+key-value-pairs. Recursively descends into each object
+and carries the correct prefix string</p>
+</dd>
+<dt><a href="#load">load(json)</a> ⇒ <code>object</code></dt>
+<dd><p>Loads a QBL in JSON notation</p>
+</dd>
+<dt><a href="#load">load(txt)</a> ⇒ <code>object</code></dt>
+<dd><p>Loads a multiline QBL expression</p>
+</dd>
+<dt><a href="#load">load(str, prefix)</a> ⇒ <code>object</code></dt>
+<dd><p>Loads a URL query string</p>
+</dd>
+</dl>
+
+<a name="query"></a>
+
+## query(input) ⇒ <code>object</code>
+Loads a Query Builder AST from generic input: either URL query string (with &)
+or as key-value-pairs (with line breaks) or as Object (from JSON or YAML notation)
+
+**Kind**: global function  
+**Returns**: <code>object</code> - Query Builder AST  
+
+| Param | Type | Description |
+| --- | --- | --- |
+| input | <code>object</code> \| <code>string</code> \| <code>Array.&lt;string&gt;</code> | QBL input |
+
+<a name="filter"></a>
+
+## filter(strings) ⇒ <code>function</code>
+Convenience function that can be used as a tagged template
+function and returns a filter function
+
+**Kind**: global function  
+**Returns**: <code>function</code> - filter function that accepts an array and returns a filtered array  
+
+| Param | Type | Description |
+| --- | --- | --- |
+| strings | <code>Array.&lt;string&gt;</code> | string array as used in template literals |
+
+<a name="unique"></a>
+
+## unique(arr, key) ⇒ <code>Array.&lt;string&gt;</code>
+Adds items to an array while ensuring there are no
+duplicate elements in the array. Can be used ao a
+reducer.
+
+**Kind**: global function  
+**Returns**: <code>Array.&lt;string&gt;</code> - the same array with unique entries  
+
+| Param | Type | Description |
+| --- | --- | --- |
+| arr | <code>Array.&lt;string&gt;</code> | an array of strings |
+| key | <code>string</code> | a string that shoul occur oly once in the arrav |
+
+<a name="flat"></a>
+
+## flat(flattened, e) ⇒ <code>array</code>
+A reducer function that flattens an array of array.
+Flattens only one level.
+
+**Kind**: global function  
+**Returns**: <code>array</code> - a flattened array  
+
+| Param | Type | Description |
+| --- | --- | --- |
+| flattened | <code>array</code> | the flattened array |
+| e | <code>array</code> | a new array to flatten |
+
+<a name="transformconjunctions"></a>
+
+## transformconjunctions(qbtree) ⇒ <code>object</code>
```

Comment:
Why not use Camelcase for function names?

Reply:
> Why not use Camelcase for function names?

he never does.... :-)